src/Processors/Formats/Impl/PrettyBlockOutputFormat.h

```diff
@@ -73,10 +73,22 @@ void registerPrettyFormatWithNoEscapesAndMonoBlock(FormatFactory & factory, cons
             const Block & sample,
             const FormatSettings & format_settings)
         {
-            if (no_escapes)
+            FormatSettings changed_settings = format_settings;
+			auto value = format_settings.pretty.output_format_pretty_color.valueOr(2);
+            switch (value)
+            {
+                case 0:
+                    changed_settings.pretty.color = false;
+                    break;
+                case 1:
+                    changed_settings.pretty.color = no_escapes ? false : true; 
+                    break;
+                case 2:
+                    changed_settings.pretty.color = isWritingToTerminal(buf) ||  no_escapes ? false : true; 
+                    break;
+            } 
+            if (!changed_settings.pretty.color)
```

Comment:
It will be better not copy and change `format_settings`. We can add a bool flag `color` to classes `PrettyBlockOutputFormat/PrettyBlockOutputFormat` constructors and use this flag instead of value from settings. So, the code here will look like this:
```
bool color = !no_escapes && format_settings.pretty.color.valueOr(isWritingToTerminal(buf));
return std::make_shared<OutputFormat>(buf, sample, format_settings, mono_block, color);
```

Reply:
Agreed, copy of `format_settings` seems unnecessary. Would we not have to change `PrettyCompactBlockOutputFormat` and `PrettySpaceBlockOutputFormat` constructors as well in this case, not only `PrettyBlockOutputFormat`?

Reply:
Yes, all `PrettyBlockOutputFormat/PrettyCompactBlockOutputFormat/PrettySpaceBlockOutputFormat`. Should not be difficult.

I just noticed that in the comment I just wrote duplicated `PrettyBlockOutputFormat/PrettyBlockOutputFormat `, I meant all these three actually :)

Reply:
I thought that was what you meant, just checking )

Reply:
@Avogar adding `color` to the constructor is easy enough, my only concern with this approach is that we mess with the consistency of the classes a little bit. For instance:

```c++
void PrettyCompactBlockOutputFormat::writeRow(
    size_t row_num,
    const Block & header,
    const Columns & columns,
    const WidthsPerColumn & widths,
    const Widths & max_widths)
{
    if (format_settings.pretty.output_format_pretty_row_numbers)
    {
        // Write row number;
        auto row_num_string = std::to_string(row_num + 1 + total_rows) + ". ";
        for (size_t i = 0; i < row_number_width - row_num_string.size(); ++i)
        {   
            writeCString(" ", out);
        }
        writeString(row_num_string, out);
    }
```
We check `output_format_pretty_row_numbers` and `max_rows` on `format_settings` object but we will check `color` as a member variable of `PrettyBlockOutputFormat`. From code cleanliness point of view probably it would be better to be consistent. 

Mind you, this inconsistency already exists in `PrettyBlockOutputFormat.cpp`, some properties we check on `format_settings` but `mono_block` is a member variable by itself. 

What do you think?

Reply:
It's totally ok to use some of the flags from `format_settings` and some from constructor. We already use it in sevaral places in formats.